src/main/java/io/cdap/plugin/gcp/bigquery/sqlengine/input/BigQuerySQLEngineAvroInputFormat.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.gcp.bigquery.sqlengine.input;
+
+import com.google.api.services.bigquery.model.Table;
+import com.google.api.services.bigquery.model.TableReference;
+import com.google.cloud.hadoop.io.bigquery.AvroBigQueryInputFormat;
+import com.google.cloud.hadoop.io.bigquery.BigQueryConfiguration;
+import com.google.cloud.hadoop.io.bigquery.BigQueryHelper;
+import com.google.cloud.hadoop.io.bigquery.Export;
+import com.google.cloud.hadoop.io.bigquery.ExportFileFormat;
+import com.google.cloud.hadoop.io.bigquery.NoopFederatedExportToCloudStorage;
+import com.google.cloud.hadoop.util.ConfigurationUtil;
+import com.google.cloud.hadoop.util.HadoopToStringUtil;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.flogger.LazyArgs.lazy;
+
+/**
+ * InputFormat which extends {@link AvroBigQueryInputFormat} and uses
+ * {@link BigQuerySQLEngineUnshardedExportToCloudStorage} as the export logic, which allows us to use Snappy to
+ * compress the output from BigQuery jobs.
+ */
+public class BigQuerySQLEngineAvroInputFormat extends AvroBigQueryInputFormat {
+
+  private static final Logger LOG = LoggerFactory.getLogger(BigQuerySQLEngineAvroInputFormat.class);
+
+  /**
+   * Copy of {@link AvroBigQueryInputFormat#getSplits(JobContext)} with a custom Export logic.
```

Comment:
can you add comments for the line(s) that are different than the copy?

Reply:
I will add these comments.